src/chunk_constraint.c

```diff
@@ -452,14 +401,17 @@ ChunkConstraints *
 ts_chunk_constraint_scan_by_chunk_id(int32 chunk_id, Size num_constraints_hint, MemoryContext mctx)
 {
 	ChunkConstraints *constraints = ts_chunk_constraints_alloc(num_constraints_hint, mctx);
-	int num_found;
+	ScanKeyData scankey[1];
+	ScanIterator iterator =
+		ts_scan_iterator_create(CHUNK_CONSTRAINT, scankey, AccessShareLock, mctx);
```

Comment:
One way to get around this issue is to make a fixed-size ScanKeyData array part of the scancontext/iterator. That would avoid having to declare the `ScanKeyData` array above, but would instead require a max-size on number of scan keys. But I think it is reasonable to assume that we can limit this to e.g., 3 or 4 keys for now and always Assert that `numkeys` is in line with that.